src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -47,14 +49,16 @@ private bool IsControlBlock(SyntaxNode node)
             }
         }
 
-        public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options, in NextGetAdjustSpacesOperation nextOperation)
+        public override AdjustSpacesOperation? GetAdjustSpacesOperation(SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options, in NextGetAdjustSpacesOperation nextOperation)
         {
+            RoslynDebug.AssertNotNull(currentToken.Parent);
+
             var operation = nextOperation.Invoke();
 
             // } else in the if else context
             if (previousToken.IsKind(SyntaxKind.CloseBraceToken)
                 && currentToken.IsKind(SyntaxKind.ElseKeyword)
-                && previousToken.Parent.Parent == currentToken.Parent.Parent)
+                && previousToken.Parent!.Parent == currentToken.Parent.Parent)
```

Comment:
interesting.  why don't you need a `!` on the RHS of the `==`.  that's somewhat surprising to me.

Reply:
I was able to make an assertion at the start of the method with respect to `currentToken.Parent`. I could make an assertion like this, but NRT wouldn't pick it up so the `!` would still be required:

```
Debug.Assert(previousToken.IsKind(SyntaxKind.None) || previousToken.Parent is object);
```